lib/cib/cib_ops.c

```diff
@@ -698,14 +700,31 @@ cib_process_xpath(const char *op, int options, const char *section,
         *answer = pcmk__xe_create(NULL, PCMK__XE_XPATH_QUERY);
     }
 
-    if (pcmk_is_set(options, cib_multiple)
-        && pcmk__str_eq(op, PCMK__CIB_REQUEST_DELETE, pcmk__str_none)) {
-        dedupXpathResults(xpathObj);
-    }
-
     for (int i = 0; i < num_results; i++) {
+        xmlNode *match = NULL;
         xmlChar *path = NULL;
-        xmlNode *match = pcmk__xpath_result(xpathObj, i);
+
+        /* If we're deleting multiple nodes, go in reverse document order.
+         * If we go in forward order and the node set contains both a parent and
+         * its descendant, then deleting the parent frees the descendent before
```

Comment:
s/descendent/descendant/
I'll correct it if there's any other feedback on this PR; otherwise I'll leave it alone.